lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1297,23 +1323,27 @@ export class UserAgentApplication {
 
         // loginRedirect
         if (stateResponse.state === this.cacheStorage.getItem(`${TemporaryCacheKeys.STATE_LOGIN}${Constants.resourceDelimiter}${stateResponse.state}`, this.inCookie) || stateResponse.state === this.silentAuthenticationState) {
+            this.logger.verbose("State matches cached state, setting requestType to login");
             stateResponse.requestType = Constants.login;
             stateResponse.stateMatch = true;
             return stateResponse;
         }
         // acquireTokenRedirect
         else if (stateResponse.state === this.cacheStorage.getItem(`${TemporaryCacheKeys.STATE_ACQ_TOKEN}${Constants.resourceDelimiter}${stateResponse.state}`, this.inCookie)) {
+            this.logger.verbose("State matches cached state, setting requestType to renewToken");
             stateResponse.requestType = Constants.renewToken;
             stateResponse.stateMatch = true;
             return stateResponse;
         }
 
         // external api requests may have many renewtoken requests for different resource
         if (!stateResponse.stateMatch) {
+            this.logger.verbose("State does not match cached state, setting requestType to type from window");
             stateResponse.requestType = window.requestType;
             const statesInParentContext = window.renewStates;
             for (let i = 0; i < statesInParentContext.length; i++) {
                 if (statesInParentContext[i] === stateResponse.state) {
+                    this.logger.verbose("StateMatch set to true");
```

Comment:
Since this is in a for loop, this may give many of the same message. Consider not having this log statement or finding a way to disambiguate between different messages.

Reply:
This looks to be the success case, and we break out of the loop when this happens. I would maybe say something like "matching state found for request", and then have a log message that is printed at the end of the loop that says "matching state not found for request".

Reply:
Changed. Let me know if there's another way you'd prefer this to look.